### PR TITLE
Add trace IDs to cached extents to help track down query gaps.

### PR DESCRIPTION
**Pull request by tomwilkie**
This is to help track down gaps in the queries.

Signed-off-by: Tom Wilkie <tom.wilkie@gmail.com>

### CODE REVIEW

File: pkg/querier/frontend/results_cache.go [hunk 1]
```diff
@@ -107,6 +112,8 @@ func (s resultsCache) handleHit(ctx context.Context, r *QueryRangeRequest, exten
 		reqResps []requestResponse
 		err      error
 	)
+	log, ctx := spanlogger.New(ctx, "handleHit")
```
Comment (gouthamve) [inline pkg/querier/frontend/results_cache.go]: Do we want to add the same line to `handleMiss`?

File: pkg/querier/frontend/frontend.proto [hunk 2]
```diff
@@ -62,5 +62,6 @@ message CachedResponse  {
 message Extent  {
 	int64 start = 1 [(gogoproto.jsontag) = "start"];
 	int64 end = 2 [(gogoproto.jsontag) = "end"];
-	APIResponse response = 3 [(gogoproto.jsontag) = "response"];
+  APIResponse response = 3 [(gogoproto.jsontag) = "response"];
```
Comment (bboreham) [inline pkg/querier/frontend/frontend.proto]: spurious change?

File: vendor/modules.txt [hunk 3]
```diff
@@ -386,12 +386,8 @@ github.com/tinylib/msgp/msgp
 # github.com/tmc/grpc-websocket-proxy v0.0.0-20190109142713-0ad062ec5ee5
 github.com/tmc/grpc-websocket-proxy/wsproxy
 # github.com/uber/jaeger-client-go v2.16.0+incompatible
-github.com/uber/jaeger-client-go/config
 github.com/uber/jaeger-client-go
-github.com/uber/jaeger-client-go/internal/baggage/remote
-github.com/uber/jaeger-client-go/internal/throttler/remote
-github.com/uber/jaeger-client-go/rpcmetrics
-github.com/uber/jaeger-client-go/transport
+github.com/uber/jaeger-client-go/config
```
Comment (bboreham) [inline vendor/modules.txt]: what happened here? looks like the file was reordered?

File: pkg/querier/frontend/results_cache.go (hunk 1 above)
Comment (tomwilkie) [reply pkg/querier/frontend/results_cache.go]: I don't think so - the miss never modifies an existing extent.  In a hit we might, so we log the old trace ids.

File: vendor/modules.txt (hunk 3 above)
Comment (tomwilkie) [reply vendor/modules.txt]: I needed this to pass lint.  No idea why.

**Pull request merged**